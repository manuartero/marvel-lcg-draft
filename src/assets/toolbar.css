#Toolbar {
  background-color: var(--medium-dark-blue);
}
