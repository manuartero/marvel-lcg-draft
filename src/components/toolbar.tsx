import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faBars } from "@fortawesome/free-solid-svg-icons";

type Props = {
  onCollection: () => void;
};

export function Toolbar({ onCollection }: Props) {
  return (
    <nav className="bg-gray-100 w-full p-2 flex items-end justify-end">
      <button className="px-8 py-4 rounded-md text-xl" onClick={onCollection}>
        <div>
          <FontAwesomeIcon icon={faBars} />
        </div>
      </button>
    </nav>
  );
}
