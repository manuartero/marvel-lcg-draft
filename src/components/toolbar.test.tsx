import { render } from "@testing-library/react";
import { Toolbar } from "./toolbar";

describe("<Toolbar />", () => {
  it("renders", () => {
    const { asFragment } = render(<Toolbar onCollection={jest.fn()} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
