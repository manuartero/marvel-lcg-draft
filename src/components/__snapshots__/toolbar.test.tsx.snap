// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Toolbar /> renders 1`] = `
<DocumentFragment>
  <nav
    class="bg-[var(--light-grey)] w-full flex items-end justify-end px-4"
    id="Toolbar"
  >
    <button
      aria-label="Deck Settings"
      class="px-8 py-4 rounded-md text-xl"
    >
      <div
        class="relative"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-dice-one "
          color="white"
          data-icon="dice-one"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M64 32C28.7 32 0 60.7 0 96V416c0 35.3 28.7 64 64 64H384c35.3 0 64-28.7 64-64V96c0-35.3-28.7-64-64-64H64zM224 224a32 32 0 1 1 0 64 32 32 0 1 1 0-64z"
            fill="currentColor"
          />
        </svg>
        <div
          class="opacity-0 absolute z-10 bg-gray-800 text-white text-sm py-1 px-2 rounded-md top-full left-1/2 transform -translate-x-1/2 show-up-transition pointer-events-none"
        >
          Deck Size
        </div>
      </div>
    </button>
    <button
      aria-label="Collection"
      class="px-8 py-4 rounded-md text-xl"
    >
      <div
        class="relative"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-list "
          color="white"
          data-icon="list"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M40 48C26.7 48 16 58.7 16 72v48c0 13.3 10.7 24 24 24H88c13.3 0 24-10.7 24-24V72c0-13.3-10.7-24-24-24H40zM192 64c-17.7 0-32 14.3-32 32s14.3 32 32 32H480c17.7 0 32-14.3 32-32s-14.3-32-32-32H192zm0 160c-17.7 0-32 14.3-32 32s14.3 32 32 32H480c17.7 0 32-14.3 32-32s-14.3-32-32-32H192zm0 160c-17.7 0-32 14.3-32 32s14.3 32 32 32H480c17.7 0 32-14.3 32-32s-14.3-32-32-32H192zM16 232v48c0 13.3 10.7 24 24 24H88c13.3 0 24-10.7 24-24V232c0-13.3-10.7-24-24-24H40c-13.3 0-24 10.7-24 24zM40 368c-13.3 0-24 10.7-24 24v48c0 13.3 10.7 24 24 24H88c13.3 0 24-10.7 24-24V392c0-13.3-10.7-24-24-24H40z"
            fill="currentColor"
          />
        </svg>
        <div
          class="opacity-0 absolute z-10 bg-gray-800 text-white text-sm py-1 px-2 rounded-md top-full left-1/2 transform -translate-x-1/2 show-up-transition pointer-events-none"
        >
          Collection
        </div>
      </div>
    </button>
  </nav>
</DocumentFragment>
`;
